packages/@lwc/template-compiler/src/parser/constants.ts

```diff
@@ -20,7 +20,7 @@ export const ITERATOR_RE = /^iterator:.*$/;
 export const EVENT_HANDLER_RE = /^on/;
 export const EVENT_HANDLER_NAME_RE = /^on[a-z][a-z0-9_]*$/;
 
-export const LWC_DIRECTIVES = {
+export const LWC_DIRECTIVES: { [type: string]: string } = {
```

Comment:
By annotating this with `{ [type: string]: string }` we are losing all the type safety on lookup:

```js
LWC_DIRECTIVES.Foo  // TypeScript considers this valid.
```

Reply:
ah ha, TIL. Will revert.